packages/dd-trace/test/appsec/rasp/command_injection.spec.js

```diff
@@ -0,0 +1,155 @@
+
+  beforeEach(() => {
+    datadogCore = {
+      storage: {
+        getStore: sinon.stub()
+      }
+    }
+
+    waf = {
+      run: sinon.stub()
+    }
+
+    utils = {
+      handleResult: sinon.stub()
+    }
+
+    commandInjection = proxyquire('../../../src/appsec/rasp/command_injection', {
+      '../../../../datadog-core': datadogCore,
+      '../waf': waf,
+      './utils': utils
+    })
+
+    config = {
+      appsec: {
+        stackTrace: {
+          enabled: true,
+          maxStackTraces: 2,
+          maxDepth: 42
+        }
+      }
+    }
+
+    commandInjection.enable(config)
+  })
+
+  afterEach(() => {
+    sinon.restore()
+    commandInjection.disable()
+  })
+
+  describe('analyzeCommandInjection', () => {
+    it('should analyze command_injection without arguments', () => {
+      const ctx = {
+        file: 'cmd',
+        shell: true
+      }
+      const req = {}
+      datadogCore.storage.getStore.returns({ req })
+
+      start.publish(ctx)
+
+      const persistent = { [addresses.SHELL_COMMAND]: 'cmd' }
+      sinon.assert.calledOnceWithExactly(waf.run, { persistent }, req, 'command_injection')
+    })
+
+    it('should analyze command_injection with arguments', () => {
+      const ctx = {
+        file: 'cmd',
+        fileArgs: ['arg0', 'arg1'],
+        shell: true
+      }
+      const req = {}
+      datadogCore.storage.getStore.returns({ req })
+
+      start.publish(ctx)
+
+      const persistent = { [addresses.SHELL_COMMAND]: ['cmd', 'arg0', 'arg1'] }
+      sinon.assert.calledOnceWithExactly(waf.run, { persistent }, req, 'command_injection')
+    })
+
+    it('should not analyze command_injection when it is not shell', () => {
+      const ctx = {
+        file: 'cmd',
+        fileArgs: ['arg0', 'arg1'],
+        shell: false
+      }
+      const req = {}
+      datadogCore.storage.getStore.returns({ req })
+
+      start.publish(ctx)
+
+      sinon.assert.notCalled(waf.run)
+    })
+
+    it('should not analyze command_injection if rasp is disabled', () => {
+      commandInjection.disable()
+      const ctx = {
+        file: 'cmd'
+      }
+      const req = {}
+      datadogCore.storage.getStore.returns({ req })
+
+      start.publish(ctx)
+
+      sinon.assert.notCalled(waf.run)
+    })
+
+    it('should not analyze command_injection if no store', () => {
+      const ctx = {
+        file: 'cmd'
+      }
+      datadogCore.storage.getStore.returns(undefined)
+
+      start.publish(ctx)
+
+      sinon.assert.notCalled(waf.run)
+    })
+
+    it('should not analyze command_injection if no req', () => {
+      const ctx = {
+        file: 'cmd'
+      }
+      datadogCore.storage.getStore.returns({})
+
+      start.publish(ctx)
+
+      sinon.assert.notCalled(waf.run)
+    })
+
+    it('should not analyze command_injection if no file', () => {
+      const ctx = {
+        fileArgs: ['arg0']
+      }
+      datadogCore.storage.getStore.returns({})
```

Comment:
There is no `req`, so the test cannot assure if it is not analyzed bc there is no file o there is no req.

Reply:
true, bad test. fixed.